src/components/nodes-table/NodeDetails.vue

```diff
@@ -0,0 +1,501 @@
+      if (!this.errorDevice) {
+        const newDevice = JSON.parse(this.deviceJSON)
+        this.socket.emit(socketActions.hass, {
+          apiName: 'add',
+          device: newDevice,
+          nodeId: this.selectedNode.id
+        })
+      }
+    },
+    apiRequest (apiName, args) {
+      if (this.socket.connected) {
+        const data = {
+          api: apiName,
+          args: args
+        }
+        this.socket.emit(socketActions.zwave, data)
+      } else {
+        this.showSnackbar('Socket disconnected')
+      }
+    },
+    async deleteDevice () {
+      const device = this.selectedDevice
+      if (
+        device &&
+        (await this.$listeners.showConfirm(
+          'Attention',
+          'Are you sure you want to delete selected device?',
+          'alert'
+        ))
+      ) {
+        this.socket.emit(socketActions.hass, {
+          apiName: 'delete',
+          device: device,
+          nodeId: this.selectedNode.id
+        })
+      }
+    },
+    async disableDiscovery () {
+      const node = this.selectedNode
+      if (
+        node &&
+        (await this.$listeners.showConfirm(
+          'Rediscover node',
+          'Are you sure you want to disable discovery of all values? In order to make this persistent remember to click on Store'
+        ))
+      ) {
+        this.socket.emit(socketActions.hass, {
+          apiName: 'disableDiscovery',
+          nodeId: this.selectedNode.id
+        })
+      }
+    },
+    exportNode () {
+      this.$listeners.export(
+        this.selectedNode,
+        'node_' + this.selectedNode.id,
+        'json'
+      )
+    },
+    getValue (v) {
+      // const node = this.nodes[v.nodeId]
+      const node = this.selectedNode
+
+      if (node && node.values) {
+        return node.values.find(i => i.id === v.id)
+      } else {
+        return null
+      }
+    },
+    async rediscoverDevice () {
+      const device = this.selectedDevice
+      if (
+        device &&
+        (await this.$listeners.showConfirm(
+          'Rediscover Device',
+          'Are you sure you want to re-discover selected device?'
+        ))
+      ) {
+        this.socket.emit(socketActions.hass, {
+          apiName: 'discover',
+          device: device,
+          nodeId: this.selectedNode.id
+        })
+      }
+    },
+    async rediscoverNode () {
+      const node = this.selectedNode
+      if (
+        node &&
+        (await this.$listeners.showConfirm(
+          'Rediscover node',
+          'Are you sure you want to re-discover all node values?'
+        ))
+      ) {
+        this.socket.emit(socketActions.hass, {
+          apiName: 'rediscoverNode',
+          nodeId: this.selectedNode.id
+        })
+      }
+    },
+    resetLocation () {
+      setTimeout(() => {
+        this.newLoc = this.selectedNode.loc
+      }, 10)
+    },
+    resetName () {
+      setTimeout(() => {
+        this.newName = this.selectedNode.name
+      }, 10)
+    },
```

Comment:
@robertsLando is there a reason why resetting is done with `setTimeout()`?
Since the node now cannot be `null` within expandible rows anymore, I'm about to simplify code as much as possible.
For name and location I'm thinking about directly binding the input fields to `node.name` / `node.loc` but this functions made me wonder if I'm overlooking something here.

Reply:
@ahochsteger I used new properties because when you edit them in this way you can restore the original ones by just clicking the clear icon, but I noticed that without that timeout the clear doesn't work reliably, sometimes it clears the input without restoring the old value

Reply:
Ok, I'll keep it.